packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SuccessTokenLongShortPairFinancialProductLibrary.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./LongShortPairFinancialProductLibrary.sol";
+import "../../../../common/implementation/Lockable.sol";
+
+/**
+ * @title Success Token Long Short Pair Financial Product Library.
+ * @notice Adds settlement logic to create success token LSPs. A success token pays out a fixed amount of
+ * collateral as a floor, with the remaining amount functioning like an embedded option. The embedded
+ * option can use the payout logic of other financial product libraries, and resemble a covered call, a
+ * KPI option, a range bond, a capped yield dollar, or custom payout logic in a new financial library.
```

Comment:
prehaps add the equation that uses the `strikePrice` `basepercentage` and `expiryPrice` to show how the `percentageLongCOllateral` is calculated in the comment here?